tests/pso_test.cpp

```diff
@@ -17,28 +17,44 @@ using namespace ens;
 using namespace ens::test;
 
 /**
- * Test the PSO optimizer on the Sphere Function.
+ * Test the PSO optimizer on the Sphere Function.  Use arma::mat.
  */
-
 TEST_CASE("LBestPSOSphereFunctionTest", "[PSOTest]")
 {
   SphereFunction f(4);
   LBestPSO s;
 
-  arma::vec coords = f.GetInitialPoint();
+  arma::mat coords = f.GetInitialPoint<arma::mat>();
   if (!s.Optimize(f, coords))
     FAIL("LBest PSO optimization reported failure for Sphere Function.");
-  
+
   double finalValue = f.Evaluate(coords);
   REQUIRE(finalValue <= 1e-5);
   for (size_t j = 0; j < 4; ++j)
     REQUIRE(coords[j] <= 1e-3);
 }
 
 /**
- * Test the PSO optimizer on the Rosenbrock Function.
+ * Test the PSO optimizer on the Sphere Function.  Use arma::fmat.
  */
+TEST_CASE("LBestPSOSphereFunctionFMatTest", "[PSOTest]")
+{
+  SphereFunction f(4);
+  LBestPSO s;
 
+  arma::fmat coords = f.GetInitialPoint<arma::fmat>();
+  if (!s.Optimize(f, coords))
+    FAIL("LBest PSO optimization reported failure for Sphere Function.");
+
+  double finalValue = f.Evaluate(coords);
+  REQUIRE(finalValue <= 1e-5);
+  for (size_t j = 0; j < 4; ++j)
+    REQUIRE(coords[j] <= 1e-3);
```

Comment:
I recommend using the [element accessor](http://arma.sourceforge.net/docs.html#element_access) with bounds check here and in other code like this, ie. `coords(j)` instead of `coords[j]`.

Since this is a test case, we should catch all possible errors.  Using `[]` allows some errors to slip through.

Reply:
Great, will apply the patch.